src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader_text.cs

```diff
@@ -3,138 +3,700 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Buffers.Text;
+using System.Diagnostics;
 using System.Runtime.CompilerServices;
 
 namespace System.Buffers.Reader
 {
     public static partial class ReaderExtensions
     {
+        // .NET parsers will typically read as many leading zeroes as are given. For a discontiguous input,
+        // (i.e. ReadOnlySequence) this behavior can be extremely expensive and/or lead to overflow and
+        // out-of-memory situations. We'll cap the maximum number of bytes to consume.
+        private const int MaxParseBytes = 128;
+        private const int MaxBoolLength = 5;
+
+        private delegate bool ParseDelegate<T>(ReadOnlySpan<byte> source, out T value, out int consumed, char standardFormat);
+        private static ParseDelegate<byte> s_byteParse;
+        private static ParseDelegate<sbyte> s_sbyteParse;
+        private static ParseDelegate<short> s_shortParse;
+        private static ParseDelegate<ushort> s_ushortParse;
+        private static ParseDelegate<int> s_intParse;
+        private static ParseDelegate<uint> s_uintParse;
+        private static ParseDelegate<long> s_longParse;
+        private static ParseDelegate<ulong> s_ulongParse;
+        private static ParseDelegate<float> s_floatParse;
+        private static ParseDelegate<double> s_doubleParse;
+        private static ParseDelegate<decimal> s_decimalParse;
+
+        private static ParseDelegate<byte> ByteParser
+        {
+            [MethodImpl(MethodImplOptions.NoInlining)]
+            get
+            {
+                if (s_byteParse == null)
+                    s_byteParse = Utf8Parser.TryParse;
+                return s_byteParse;
+            }
+        }
+
+        private static ParseDelegate<sbyte> SbyteParser
```

Comment:
nit: I would call it "SByte" to match the type name

Reply:
I was matching the c# keywords. I can change these to Single etc.